`Unknown login type m.login.password` since 1.8.0

My homeserver is configured to only support bearer token auth and (non-native) OIDC.
After I upgraded from 1.7.0 to 1.8.0, without changing the bot's config, the bot did not start anymore, and I was greeted with the error `{ errcode: 'M_UNKNOWN', error: 'Unknown login type m.login.password' }`. I've tried adding `encryption.use: false` (which probably is the default anyway) unsuccessfully. The only auth related thing configured is `accessToken`.


```log
Oct 02 07:28:48 selene systemd[1]: Started matrix-bot-mjolnir.service - Matrix Mjolnir bot.
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: 2024-10-02 05:28:51.560179: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: Wed, 02 Oct 2024 05:28:51 GMT [INFO] [index] Starting bot...
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: Wed, 02 Oct 2024 05:28:51 GMT [INFO] [Healthz] Listening for health requests on 0.0.0.0:8080
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: Wed, 02 Oct 2024 05:28:51 GMT [ERROR] [MatrixHttpClient] (REQ-1) { errcode: 'M_UNKNOWN', error: 'Unknown login type m.login.password' }
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: Failed to setup mjolnir from the config /data: Error: M_UNKNOWN: Unknown login type m.login.password
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: /node_modules/@vector-im/matrix-bot-sdk/lib/http.js:10
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:         new MatrixError_1.MatrixError(errBody, response.statusCode, response.headers) : undefined;
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:         ^
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: MatrixError: M_UNKNOWN: Unknown login type m.login.password
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:     at Object.defaultErrorHandler [as errorHandler] (/node_modules/@vector-im/matrix-bot-sdk/lib/http.js:10:9)
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:     at doHttpRequest (/node_modules/@vector-im/matrix-bot-sdk/lib/http.js:98:31)
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:     at async descriptor.value (/node_modules/@vector-im/matrix-bot-sdk/lib/metrics/decorators.js:19:32)
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:     at async MatrixAuth.passwordLogin (/node_modules/@vector-im/matrix-bot-sdk/lib/MatrixAuth.js:114:26)
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:     at async /mjolnir/index.js:85:36 {
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:   body: {
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:     errcode: 'M_UNKNOWN',
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:     error: 'Unknown login type m.login.password'
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:   },
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:   statusCode: 400,
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:   errcode: 'M_UNKNOWN',
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:   error: 'Unknown login type m.login.password',
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]:   retryAfterMs: undefined
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: }
Oct 02 07:28:51 selene matrix-bot-mjolnir[1394658]: Node.js v20.17.0
```

--- a/src/index.ts
+++ b/src/index.ts
@@ -69,7 +69,7 @@ import { initializeSentry, initializeGlobalPerformanceMetrics, patchMatrixClient
         if (config.pantalaimon.use) {
             const pantalaimon = new PantalaimonClient(config.homeserverUrl, storage);
             client = await pantalaimon.createClientWithCredentials(config.pantalaimon.username, config.pantalaimon.password);
-        } else {
+        } else if (config.encryption.use) {
             const accessToken = await Promise.resolve(storage.readValue("access_token"));
             if (accessToken) {
                 client = new MatrixClient(config.homeserverUrl, accessToken, storage, cryptoStorage);
@@ -78,7 +78,6 @@ import { initializeSentry, initializeGlobalPerformanceMetrics, patchMatrixClient
                 const tempClient = await auth.passwordLogin(config.encryption.username, config.encryption.password)
                 client = new MatrixClient(config.homeserverUrl, tempClient.accessToken, storage, cryptoStorage);
             }
-
             try {
                 LogService.info("index", "Preparing encrypted client...")
                 await client.crypto.prepare();
@@ -86,8 +85,8 @@ import { initializeSentry, initializeGlobalPerformanceMetrics, patchMatrixClient
                 LogService.error("Index", `Error preparing encrypted client ${e}`)
                 throw e
             }
-
-
+        } else {
+            client = new MatrixClient(config.homeserverUrl, config.accessToken, storage);
         }
         patchMatrixClient();
         config.RUNTIME.client = client;

